pkg/k8s/factory_functions.go

```diff
@@ -402,6 +436,52 @@ func ConvertToCNPWithStatus(obj interface{}) interface{} {
 	}
 }
 
+// ConvertToCCNP converts a *cilium_v2.CiliumClusterwideNetworkPolicy into a
+// *types.SlimCNP without the Status field of the given CNP, or a
+// cache.DeletedFinalStateUnknown into a cache.DeletedFinalStateUnknown with a
+// *types.SlimCNP, also without the Status field of the given CNP, in its Obj.
+// If the given obj can't be cast into either *cilium_v2.CiliumClusterwideNetworkPolicy
+// nor cache.DeletedFinalStateUnknown, the original obj is returned.
+// WARNING calling this function will set *all* fields of the given CNP as
+// empty.
+func ConvertToCCNP(obj interface{}) interface{} {
+	switch concreteObj := obj.(type) {
+	case *cilium_v2.CiliumClusterwideNetworkPolicy:
+		cnp := &types.SlimCNP{
+			CiliumNetworkPolicy: &cilium_v2.CiliumNetworkPolicy{
+				TypeMeta:   concreteObj.TypeMeta,
+				ObjectMeta: concreteObj.ObjectMeta,
+				Spec:       concreteObj.Spec,
+				Specs:      concreteObj.Specs,
+			},
+		}
+		*concreteObj = cilium_v2.CiliumClusterwideNetworkPolicy{}
+		return cnp
+
+	case cache.DeletedFinalStateUnknown:
+		cnp, ok := concreteObj.Obj.(*cilium_v2.CiliumClusterwideNetworkPolicy)
+		if !ok {
+			return obj
+		}
+		dfsu := cache.DeletedFinalStateUnknown{
+			Key: concreteObj.Key,
+			Obj: &types.SlimCNP{
+				CiliumNetworkPolicy: &cilium_v2.CiliumNetworkPolicy{
+					TypeMeta:   cnp.TypeMeta,
+					ObjectMeta: cnp.ObjectMeta,
+					Spec:       cnp.Spec,
+					Specs:      cnp.Specs,
+				},
+			},
+		}
+		*cnp = cilium_v2.CiliumClusterwideNetworkPolicy{}
+		return dfsu
+
+	default:
+		return obj
+	}
```

Comment:
it looks to me that status is missing, no?

Reply:
No, there are actually two methods `ConvertToCCNP` and `ConvertToCCNPWithStatus`. The first one converts the object to types.SlimCNP without the status field, therefore, there is no status here.